### PR TITLE
feat: init server side gtm template

### PR DESCRIPTION
**Pull request by yuhao900914**
# Amplitude Developer Docs PR

## Description

https://pr-784.d19s7xzcva2mw3.amplifyapp.com/data/sources/google-tag-manager-server/

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sources/google-tag-manager-server.md [hunk 1]
```diff
@@ -8,10 +8,144 @@ This is the server-side Google Tag Manager Template for Amplitude Analytics. The
...
+If you start from zero, you need to setup your conatiner first. This Amplitude Analytics tag template can be found in **Server** target platform which for the uses on desktop and mobile web pages.
+
+![Server Container Setup](../../assets/images/gtm/gtm-server-conatiner-setup.png)
+
+### Add Template
+
+Create a new tag template by searching the gallery. Choose Amplitude Analytics and click `add` button.
+![Amplitude Analytics](../../assets/images/gtm/gtm-amplitude-analytics.png)
+
+### Create a Client
```
Comment (Mercy811) [inline docs/data/sources/google-tag-manager-server.md:27]: Feel like [this](https://developers.google.com/tag-platform/learn/sst-fundamentals/2-what-is-sst) is really helpful to understand how server-side GTM works. And why clients are needed. Maybe have this link here?

**Review by Mercy811 — APPROVED** [review]
Thank you Alyssa for making this up. Not very familiar with server-side template as I've never been asked questions about it. LGTM and we can add more content later if there is any server-side GTM question.

**Review by justin-fiedler — APPROVED** [review]
Thanks @yuhao900914 this is great.

Minor nit: I think example screenshots of GTM configuration would go a long way. Can you add some pictures?

**Comment by yuhao900914** [comment]
Added more screenshots.

**Pull request merged**